### PR TITLE
EE_Venue type fixes from cafe

### PR DESCRIPTION
**Pull request by Pebblo**
Fixes fatal such as:

`
2023/01/18 15:59:52 [error] 4514#4514: *7523 FastCGI sent in stderr: "PHP message: PHP Fatal error: Uncaught TypeError: EE_Venue::country_ID(): Return value must be of type string, null returned in /event-espresso-core-reg/core/db_classes/EE_Venue.class.php:204
Stack trace:
#0 /event-espresso-core-reg/core/helpers/EEH_Address.helper.php(102): EE_Venue->country_ID()
#1 /event-espresso-core-reg/core/helpers/EEH_Address.helper.php(48): EEH_Address::_regular_formatting(Object(EventEspresso\core\services\address\formatters\InlineAddressFormatter), Object(EE_Venue), false)
#2 /event-espresso-core-reg/core/helpers/EEH_Venue_View.helper.php(345): EEH_Address::format(Object(EE_Venue), 'inline', false, false)
#3 /event-espresso-core-reg/public/template_tags.php(1312): EEH_Venue_View::venue_has_address(72142)<!-- 
`

### CODE REVIEW

**Review by knazart — COMMENTED** [review]
Looks ok to me 👍  
BUT, we now work in the Cafe repo, so these changes need to occur there. And some of the changes might already be in Cafe, might.

File: core/db_classes/EE_Venue.class.php [hunk 1]
```diff
@@ -231,21 +273,23 @@ public function country_obj(): ?EE_Country
...
     public function country(): string
     {
-        if (apply_filters('FHEE__EEI_Address__country__use_abbreviation', true, $this->country_obj())) {
-            return $this->country_ID();
-        } else {
-            return $this->country_name();
-        }
+        return apply_filters('FHEE__EEI_Address__country__use_abbreviation', true, $this->country_obj())
+            ? $this->country_ID()
+            : $this->country_name();
```
Comment (hoseinrafiei) [inline core/db_classes/EE_Venue.class.php:283]: 👍

**Review by hoseinrafiei — APPROVED** [review]
Looks good.
I believe we already have most of the changes in Cafe repo. I think you should work on Cafe repo instead of this repo.

**Comment by Pebblo** [comment]
Thank you @knazart @hoseinrafiei 

These changes **are** from Cafe (EE5), I copied the above file from cafe and into 'core' (EE4)

I can't push a release from Cafe currently and if I get reports of issues from users using core I need to either bring over the changes from cafe (as in this PR) or if the issue is in both, apply them to cafe and then bring them into here so I can build a current release but thank you for checking to make sure I wasn't just working on the wrong repo :)

Merging this in ready for release.

**Pull request merged**